crates/consensus/src/contract/mod.rs

```diff
@@ -0,0 +1,254 @@
+        let t1_idx = asset_builder.add_or_get_asset(t1) as u16;
+        // Build our Pair featuring our uniform clearing price
+        // This price is in Ray format as requested.
+        let uniswap_price: U256 = *solution.ucp;
+        // pairs.push(Pair { t0_idx, t1_idx, uniswap_price });
+        let pair_idx = pairs.len() - 1;
+
+        // Pull out our net AMM order
+        let net_amm_order = solution
+            .amm_quantity
+            .as_ref()
+            .map(|amm_o| amm_o.to_order_tuple(t0_idx, t1_idx));
+        // Pull out our TOB swap and TOB reward
+        let (tob_swap, tob_rewards) = solution
+            .searcher
+            .as_ref()
+            .map(|tob| {
+                let swap = if tob.is_bid {
+                    (t1_idx, t0_idx, tob.quantityIn, tob.quantityOut)
+                } else {
+                    (t0_idx, t1_idx, tob.quantityIn, tob.quantityOut)
+                };
+                // let amm = MarketSnapshot::new(vec![], SqrtPriceX96::default()).unwrap();
+                //let rewards = calculate_reward(tob, amm).unwrap();
+                let rewards = ToBOutcome::default();
+                (Some(swap), Some(rewards))
+            })
+            .unwrap_or_default();
+        // Merge our net AMM order with the TOB swap
+        let merged_amm_swap = match (net_amm_order, tob_swap) {
+            (Some(amm), Some(tob)) => {
+                if amm.0 == tob.0 {
+                    // If they're in the same direction we just sum them
+                    Some((amm.0, amm.1, (amm.2 + tob.2), (amm.3 + tob.3)))
+                } else {
+                    // If they're in opposite directions then we see if we have to flip them
+                    if tob.2 > amm.3 {
+                        Some((tob.0, tob.1, tob.2 - amm.2, tob.3 - amm.3))
+                    } else {
+                        Some((amm.0, amm.1, amm.2 - tob.3, amm.3 - tob.2))
+                    }
+                }
+            }
+            (net_amm_order, tob_swap) => net_amm_order.or(tob_swap)
+        };
+        // Unwrap our merged amm order or provide a zero default
+        let (asset_in_index, asset_out_index, quantity_in, quantity_out) =
+            merged_amm_swap.unwrap_or((t0_idx, t1_idx, 0_u128, 0_u128));
+        // If we don't have a rewards update, we insert a default "empty" struct
+        let tob_outcome = tob_rewards.unwrap_or_default();
+
+        // Account for our net AMM Order
+        asset_builder.uniswap_swap(
+            AssetBuilderStage::Swap,
+            asset_in_index as usize,
+            asset_out_index as usize,
+            quantity_in,
+            quantity_out
+        );
+        // Account for our reward
+        asset_builder.allocate(AssetBuilderStage::Reward, t0, tob_outcome.total_reward.to());
+        let rewards_update = tob_outcome.to_rewards_update();
+        // Push the pool update
+        pool_updates.push(PoolUpdate {
+            zero_for_one: false,
```

Comment:
doesn't this have b calculated by checked the asset indexes?

Reply:
not my code